### PR TITLE
Document Qt5 build and install for Linux/BSD

### PR DESCRIPTION
**Pull request by YakoYakoYokuYoku**
Thanks for submitting a pull request! Please provide enough information so that others can review your pull request. Additionally, make sure you've done all of these things:

- [x] I've followed the [contributing guidelines](https://github.com/NatronGitHub/Natron/blob/RB-2.4/CODE_OF_CONDUCT.md) to the best of my understanding
- [x] I've read and understood the [contributing guidelines](https://github.com/NatronGitHub/Natron/blob/RB-2.4/CONTRIBUTING.md)
- [x] I've formatted my code according to Natron's [code style]([#](https://github.com/NatronGitHub/Natron#logistics))
- [x] I've searched the [pull requests tracker](https://github.com/NatronGitHub/Natron/pulls?q=is%3Apr) to ensure that this PR is not a duplicate

## PR Description

**What type of PR is this? (Check one of the boxes below)**

- [x] My change requires a change to the documentation
    - [x] I have updated the documentation accordingly

**What does this pull request do?**

This was due job from the Qt5 pulls, I've priotized Qt5 over Qt4 for obvious reasons and in the hopes that more people can test the `RB-2.5` branch.

**Show a few screenshots (if this is a visual change)**

N/A.

**Have you tested your changes (if applicable)? If so, how?**

Build Natron according to the doc.

**Futher details of this pull request**

~I'll try to work on the BSD documentation as it's very similar to Linux~ done. For the other two platforms, Windows and MacOS, I don't have the sufficient knowledge about it so other devs are invited to modify their instructions.

### CODE REVIEW

**Comment by rodlie** [comment]
Looks ok, but why remove CentOS? the installation instructions for Fedora and CentOS are the same.

* RHEL/CentOS 7 has Qt 5.12 (and 4.8.7)
* RHEL/CentOS 8 had Qt 5.12 (may have changed since last time I used it)
* RHEL/CentOS 9 has Qt 5.15(.2+patches)

**Comment by YakoYakoYokuYoku** [comment]
> Looks ok, but why remove CentOS? the installation instructions for Fedora and CentOS should be the same.
> 
> - RHEL/CentOS 7 has Qt 5.12 (and 4.8.7)
> - RHEL/CentOS 9 has Qt 5.15(.2+patches)

CentOS was removed from the documentation due to [reaching its EOL](https://www.centos.org/centos-linux-eol/) on 2021 New Year's Eve. We could point users to [Rocky Linux](https://rockylinux.org/) instead, but I think they'll figure out that they could follow Fedora instructions.

**Comment by rodlie** [comment]
> CentOS was removed from the documentation due to [reaching its EOL](https://www.centos.org/centos-linux-eol/) on 2021 New Year's Eve.

* CentOS Linux 7 is still maintained and is the current vfxplatform target
* CentOS Stream 8 is still maintained and will probably be the vfxplatform target next year

File: INSTALL_LINUX.md [hunk 1]
```diff
@@ -383,23 +407,17 @@ pyside {
...
-
-Instructions for CentOS and Fedora.
+## Fedora
 
-On CentOS you need the EPEL repository:
-
-```
-yum install epel-release
-```
+Instructions for Fedora.
```
Comment (rodlie) [inline INSTALL_LINUX.md]: Not needed, Fedora and RHEL share the same package names

File: INSTALL_LINUX.md [hunk 2]
```diff
@@ -383,23 +407,17 @@ pyside {
 }
 ```
 
-## CentOS7
-
-Instructions for CentOS and Fedora.
+## Fedora
 
-On CentOS you need the EPEL repository:
```
Comment (rodlie) [inline INSTALL_LINUX.md:390]: Keep this and add
````
On RHEL (and compatible) ...
````
?

File: INSTALL_LINUX.md [hunk 3]
```diff
@@ -383,23 +407,17 @@ pyside {
 }
 ```
 
-## CentOS7
-
-Instructions for CentOS and Fedora.
+## Fedora
```
Comment (rodlie) [inline INSTALL_LINUX.md]: Maybe
```
## Fedora/RHEL (and compatible)
```
?

File: INSTALL_LINUX.md [hunk 4]
```diff
@@ -30,7 +33,7 @@ This file is supposed to guide you step by step to have working (compiling) vers
 4. [Distribution specific](#distribution-specific)
     - [Arch Linux](#arch-linux)
     - [Debian-based](#debian-based)
-    - [CentOS7/Fedora](#centos7)
+    - [Fedora](#fedora)
```
Comment (rodlie) [inline INSTALL_LINUX.md]: We could maybe use
```
[Fedora/RHEL](#fedora)
```
?

**Comment by YakoYakoYokuYoku** [comment]
I've been looking at the [VFX Reference Platform site](https://vfxplatform.com/) but it doesn't mention anything about CentOS being a target. I'll modify the documentation to point at Fedora/RHEL and derivatives to at least clear the situation for Enterprise Linux.

**Comment by devernay** [comment]
The reason for picking CentOS as the build platform is that it's the only platform (to the best of my knowledge) that allows building binaries that can run on every currently supported Linux distrib.

Through the [Redhat Developer Toolset](https://developers.redhat.com/products/developertoolset/overview), we get modern tool versions, including GCC, and the binaries can run even on a Linux that has a centos7-era libstdc++ and libgcc. It's using lots of tricks to do so, which are probably documented somewhere.

This is the reason why it's the reference *build* platform for Natron, and it's probably the same platform that every software company in the VXF reference platform consortium (which we are not part of) uses. Take a look at every software vendor requirements: what's the oldest supported Linux distribution? CentOS 7.

See:
- [Nuke requirements](https://www.foundry.com/products/nuke-family/requirements)
- [Maya requirements](https://knowledge.autodesk.com/support/maya/troubleshooting/caas/sfdcarticles/sfdcarticles/System-requirements-for-Autodesk-Maya-2022.html)
- [Houdini requirements](https://www.sidefx.com/Support/system-requirements/) - they tell you to use GCC 9.3.1, which is... the compiler in Redhat DTS9
- etc.

**Comment by YakoYakoYokuYoku** [comment]
> The reason for picking CentOS as the build platform is that it's the only platform (to the best of my knowledge) that allows building binaries that can run on every currently supported Linux distrib.
> 
> Through the [Redhat Developer Toolset](https://developers.redhat.com/products/developertoolset/overview), we get modern tool versions, including GCC, and the binaries can run even on a Linux that has a centos7-era libstdc++ and libgcc. It's using lots of tricks to do so, which are probably documented somewhere.
> 
> This is the reason why it's the reference _build_ platform for Natron, and it's probably the same platform that every software company in the VXF reference platform consortium (which we are not part of) uses. Take a look at every software vendor requirements: what's the oldest supported Linux distribution? CentOS 7.

Will take into consideration when using GitHub actions for releases.

On another note there's #803 that removes the step of running Shiboken manually.

**Review by rodlie — APPROVED** [review]

**Pull request merged**